src/main/java/sirius/kernel/timer/EndOfDayTaskExecutor.java

```diff
@@ -0,0 +1,168 @@
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.IntPredicate;
+
+/**
+ * Executes all {@link EndOfDayTask end of day tasks} in the given timeframe.
+ *
+ * @see EndOfDayTask for a detailed description of the theory of operations
+ */
+@Register(classes = {EndOfDayTaskExecutor.class, EveryDay.class})
+public class EndOfDayTaskExecutor implements EveryDay {
+
+    @ConfigValue("timer.daily.end-of-day")
+    private int startHour;
+
+    @ConfigValue("health.endOfDayTasksLimit")
+    private int endHour;
+
+    @Part
+    private Tasks tasks;
+
+    @Part
+    private TimeProvider timeProvider;
+
+    @Parts(EndOfDayTask.class)
+    private PartCollection<EndOfDayTask> endOfDayTasks;
+
+    private Map<EndOfDayTask, EndOfDayTaskInfo> taskInfos = new ConcurrentHashMap<>();
+
+    @Override
+    public String getConfigKeyName() {
+        return "end-of-day";
+    }
+
+    @Override
+    public void runTimer() throws Exception {
+        tasks.defaultExecutor().start(this::executeTasks);
+    }
+
+    private void executeTasks() {
+        // As we limit the execution to a certain time frame we shuffle the execution on each run
+        // so that no single task can steal all the processing time...
+        List<EndOfDayTask> tasksToExecute = new ArrayList<>(endOfDayTasks.getParts());
+        Collections.shuffle(tasksToExecute);
+
+        IntPredicate endCondition = determineEndCondition();
+        for (EndOfDayTask task : tasksToExecute) {
+            int currentHour = timeProvider.localTimeNow().getHour();
+            if (endCondition.test(currentHour)) {
+                Log.BACKGROUND.INFO("Aborting end of day tasks, as the time limit has been hit...");
+                return;
+            }
+
+            Log.BACKGROUND.INFO("Executing end of day task: %s", task.getName());
+            EndOfDayTaskInfo info = executeTask(task);
+            Log.BACKGROUND.INFO("Executed %s - Took: %s, Success: %s, Last Error: %s",
+                                task.getName(),
+                                info.getFormattedLastDurartion(),
+                                info.isLastExecutionWasSuccessful(),
+                                info.getLastErrorMessage());
+        }
+    }
+
+    /**
+     * Determines if processing should keep running.
+     * <p>
+     * If the <tt>endHour</tt> is greater than the start hour, we stop processing as soon as the current hour is past
+     * the end hour. Otherwise (which is probably the most common case) we stop processing if the current hour is past
+     * the end hour but less thant the original start hour.
+     *
+     * @return a condition which yields <tt>true</tt> as soon as processing should be aborted
+     */
+    private IntPredicate determineEndCondition() {
+        if (startHour < endHour) {
+            return currentHour -> currentHour > endHour;
```

Comment:
```suggestion
            return currentHour -> currentHour > endHour && currentHour < 24 || currentHour < startHour && currentHour >= 0;
```
![Bildschirmfoto 2020-03-19 um 13 34 06](https://user-images.githubusercontent.com/60612914/77068176-71270a80-69e6-11ea-9a3a-049b8473a37b.png)

Reply:
Pushed another - hopefully simpler - approach

Reply:
Looks good!